core/src/main/java/org/owasp/dependencycheck/analyzer/PinnedMavenInstallAnalyzer.java

```diff
@@ -0,0 +1,262 @@
+     */
+    private static final Pattern MAVEN_INSTALL_JSON_PATTERN = Pattern.compile("(.+install.*|.*install.+)\\.json");
+
+    /**
+     * Match any files that look like *install*.json.
+     */
+    private static final FileFilter FILTER = (File file) -> MAVEN_INSTALL_JSON_PATTERN.matcher(file.getName()).matches();
+
+    @Override
+    protected FileFilter getFileFilter() {
+        return FILTER;
+    }
+
+    @Override
+    public String getName() {
+        return ANALYZER_NAME;
+    }
+
+    @Override
+    public AnalysisPhase getAnalysisPhase() {
+        return ANALYSIS_PHASE;
+    }
+
+    @Override
+    protected String getAnalyzerEnabledSettingKey() {
+        return Settings.KEYS.ANALYZER_MAVEN_INSTALL_ENABLED;
+    }
+
+    @Override
+    protected void analyzeDependency(Dependency dependency, Engine engine) throws AnalysisException {
+        LOGGER.debug("Checking file {}", dependency.getActualFilePath());
+
+        final File dependencyFile = dependency.getActualFile();
+        if (!dependencyFile.isFile() || dependencyFile.length() == 0) {
+            return;
+        }
+
+        DependencyTree tree;
+        try {
+            InstallFile installFile = INSTALL_FILE_READER.readValue(dependencyFile);
+            tree = installFile.dependencyTree;
+        } catch (IOException e) {
+            return;
+        }
+
+        if (tree == null) {
+            return;
+        } else if (!Objects.equals(tree.autogeneratedSentinel, "THERE_IS_NO_DATA_ONLY_ZUUL")) {
+            return;
+        }
+
+        engine.removeDependency(dependency);
+
+        if (!Objects.equals(tree.version, "0.1.0")) {
+            LOGGER.warn("Unsupported pinned maven_install.json version {}. Continuing optimistically.", tree.version);
+        }
+
+        List<MavenDependency> deps = tree.dependencies;
+        if (deps == null) {
+            deps = Collections.emptyList();
+        }
+
+        for (MavenDependency dep : deps) {
+            if (dep.coord == null) {
+                LOGGER.warn("Unexpected null coordinate in {}", dependency.getActualFilePath());
+                continue;
+            }
+
+            LOGGER.debug("Analyzing {}", dep.coord);
+            String[] pieces = dep.coord.split(":");
+            if (pieces.length < 3 || pieces.length > 5) {
+                LOGGER.warn("Invalid maven coordinate {}", dep.coord);
+                continue;
+            }
+
+            String group = pieces[0];
+            String artifact = pieces[1];
+            String version;
+            String classifier = null;
+            if (pieces.length == 3) {
+                version = pieces[2];
+            } else if (pieces.length == 4) {
+                classifier = pieces[2];
+                version = pieces[3];
+            } else {
+                // length == 5 as guaranteed above.
+                classifier = pieces[3];
+                version = pieces[4];
+            }
+
+            if ("sources".equals(classifier)) {
```

Comment:
likely rare - but if we are skipping `sources` we should also skip `javadoc`:

```suggestion
            if ("sources".equals(classifier) || "javadoc".equals(classifier)) {
```